src/lib/selections.ts

```diff
@@ -455,25 +459,17 @@ export type ResolvedSelectionType = [Selection['type'] | 'other', number]
 export function getSelectionType(
   selection: Selections
 ): ResolvedSelectionType[] {
-  return selection.codeBasedSelections
-    .map((s, i) => {
-      if (canExtrudeSelectionItem(selection, i)) {
-        return ['extrude-wall', 1] as ResolvedSelectionType // This is implicitly determining what a face is, which is bad
-      } else {
-        return ['other', 1] as ResolvedSelectionType
-      }
-    })
-    .reduce((acc, [type, count]) => {
-      const foundIndex = acc.findIndex((item) => item && item[0] === type)
-
-      if (foundIndex === -1) {
-        return [...acc, [type, count]]
-      } else {
-        const temp = [...acc]
-        temp[foundIndex][1] += count
-        return temp
-      }
-    }, [] as ResolvedSelectionType[])
+  const extrudableCount = selection.codeBasedSelections.filter((_, i) => {
+    const singleSelection = {
+      ...selection,
+      codeBasedSelections: [selection.codeBasedSelections[i]],
+    }
```

Comment:
This is likely a nit, but aren't you already set up to isolate the `codeBasedSelections` up [within the `canExtrudeSelectionItem` function itself](https://github.com/KittyCAD/modeling-app/pull/3884/files#diff-cbfdb1c1f91506eba7d5317cd631d1ce31769080940fa9d5d525940c0c708132R436-R439)?

Reply:
nah, can extrude selection gets as an argument the whooooole list of selections and an index of the one to check. While isSketchPile is expecting only one argument - selection. Not selection+index. So it gets the whole array of selections and just fails to work with it. 
<img width="667" alt="Screenshot 2024-09-17 at 22 19 10" src="https://github.com/user-attachments/assets/0f70a176-4ba5-4703-80f5-a0c0cd1d3de3">

Reply:
> we should rethink the generic approach in future.
@franknoirot 
agree, those are good points, probably will be covered by https://github.com/KittyCAD/modeling-app/issues/3837